code/datums/components/transforming.dm

```diff
@@ -0,0 +1,259 @@
+/*
+ * Transforming weapon component. For weapons that swap between states.
+ * For example: Energy swords, cleaving saws, switch blades.
+ *
+ * Used to easily make an item that can be attack_self'd to gain force or change mode.
+ *
+ * Only values passed on initialize will update when the item is activated (except the icon_state).
+ * The icon_state of the item will swap between "[icon_state]" and "[icon_state]_on".
+ */
+/datum/component/transforming
+	/// Whether the weapon is transformed
+	var/active = FALSE
+	/// Cooldown on transforming this item back and forth
+	var/transform_cooldown_time
+	/// Force of the weapon when active
+	var/force_on
+	/// Throwforce of the weapon when active
+	var/throwforce_on
+	/// Throw speed of the weapon when active
+	var/throw_speed_on
+	/// Throw speed of the weapon when active
+	var/bleedforce_on
+	/// Weight class of the weapon when active
+	var/w_class_on
+	/// The sharpness of the weapon when active
+	var/sharpness_on
+	/// Hitsound played when active
+	var/hitsound_on
+	/// List of the original continuous attack verbs the item has.
+	var/list/attack_verb_continuous_off
+	/// List of the original simple attack verbs the item has.
+	var/list/attack_verb_simple_off
+	/// List of continuous attack verbs used when the weapon is enabled
+	var/list/attack_verb_continuous_on
+	/// List of simple attack verbs used when the weapon is enabled
+	var/list/attack_verb_simple_on
+	/// Whether clumsy people need to succeed an RNG check to turn it on without hurting themselves
+	var/clumsy_check
+	/// If we get sharpened with a whetstone, save the bonus here for later use if we un/redeploy
+	var/sharpened_bonus = 0
+	/// Cooldown in between transforms
+	COOLDOWN_DECLARE(transform_cooldown)
+
+/datum/component/transforming/Initialize(
+		start_transformed = FALSE,
+		transform_cooldown_time = 0 SECONDS,
+		force_on = 0,
+		throwforce_on = 0,
+		throw_speed_on = 2,
+		bleedforce_on = NONE,
```

Comment:
We default all bleedforce to NONE(like `obj/item`, since very few actual transforming weapons (maybe 5?) use greater bleeding values

Reply:
None is a define specifically for bitmasks, whereas bleeding is just a standard integer value.

```suggestion
		bleedforce_on = 0,
```

Reply:
ded0212